src/control/mod.rs

```diff
@@ -936,6 +941,35 @@ pub trait Device: super::Device {
         Ok(())
     }
 
+    /// Create a drm lease
+    fn create_lease(
+        &self,
+        objects: &[RawResourceHandle],
+        flags: OFlag,
+    ) -> Result<(DrmLeaseId, RawFd), SystemError> {
```

Comment:
Would separating the lease ID and underlying file descriptor be a good idea? Or wrapping them up into something that takes ownership to be dropped later/not outlive the underlying device?

Reply:
Usually the file descriptor will be send to another process, while the lease_id will be kept in the current process to keep track of the lease and revoke it later.

So I guess we could technically wrap the `LeaseId`, so that it gets revoked on drop, but that needs access to the `control::Device` used to create, which potentially introduces lifetime issues. This would also be different compared to most other objects this crate crates, e.g. framebuffers, which are not cleaned up on drop.

But I guess we could wrap the `RawFd` in an `OwnedFd`, so it cannot be kept open accidentally, I will fix that.

Reply:
Ah, got it. The file descriptor gets sent to another process via UNIX socket or something? (I'm guessing an X11 or Wayland protocol?)

In that case, the `OwnedFd` is probably good.

Reply:
Yes exactly. Leasing basically lets you borrow a subset of drm resources into a new file descriptor without giving up DRM master.